### ISSUE TITLE
`unnecessary-comprehension` false positive

### ISSUE THREAD

**Issue by nickdrozd** [pylint-dev/pylint#8577]
### Bug description

```python
a = [[1, 2], [3, 4]]
b = [(x, y) for x, y in a]  # flagged as unnecessary-comprehension
```

### Command used

```shell
pylint asdf.py
```

### Pylint output

```shell
asdf.py:2:4: R1721: Unnecessary use of a comprehension, use a instead. (unnecessary-comprehension)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 3.0.0b1
astroid 2.16.0dev0
Python 3.11.0
```

**Comment by jacobtylerwalls** [comment]
Because it does something useful by converting the inner lists to tuples. Also "use a instead" isn't right...
